actinia/mapset.py

```diff
@@ -23,31 +23,219 @@
+            (lock) route:
+                /locations/{location_name}/mapsets/{mapset_name}/raster_layers
+            (raster_layers) route:
+                /locations/{location_name}/mapsets/{mapset_name}/raster_layers
+            (vector_layers) route:
+                /locations/{location_name}/mapsets/{mapset_name}/vector_layers
+            (strds) route:
+                /locations/{location_name}/mapsets/{mapset_name}/strds - GET
+            (processing) route:
+                /locations/{location_name}/mapsets/{mapset_name}/processing
+            (processing_async) route:
+                /locations/{location_name}/mapsets/{mapset_name}/processing_async
+
+        raster_layers : bool
+        Returns
+        -------
+        base_url : str
+            Return the url scheme for the mapset request
+        """
+        base_url = f"{actinia_url}/locations/{location_name}/mapsets"
+        if mapset_name is not None:
+            base_url = f"{base_url}/{mapset_name}"
+            if isinstance(task, MAPSET_TASK):
+                base_url = f"{base_url}/{mapset_name}{task.value}"
+
+        return base_url
+
+    def info(self):
+        """Get mapset info"""
+        if self.projection is None or self.region is None:
+            proc_res = self.request_info(
+                self.name,
+                self.__location_name,
+                self.__actinia,
+                self.__auth
+            )
+            self.projection = proc_res["projection"]
+            self.region = Region(**proc_res["region"])
+        return {"projection": self.projection, "region": self.region}
+
+    def delete(self):
+        """Deletes the mapset"""
+        self.delete_mapset_request(
+            self.name,
+            self.__location_name,
+            self.__actinia,
+            self.__auth
+        )
+        del self.__actinia.locations[self.__location_name].mapsets[self.name]
+
+    @classmethod
+    def list_mapsets_request(cls, location_name, actinia, auth):
+        """
+        Creates a mapset within a location.
+
+        Parameters
+        ----------
+        location_name : str
+            The name of the location where the mapsets are located.
+        actinia : Actinia
+            An Actinia instance containing the url
+        auth :
+            Actinia authentication
+        Returns
+        -------
+        Mapset
+            A new mapset instance for the created mapset
+        """
+        url = cls.__request_url(actinia.url, location_name)
+        resp = requests.get(url, auth=auth)
+        if resp.status_code != 200:
+            raise Exception(f"Error {resp.status_code}: {resp.text}")
+
+        mapset_names = json.loads(resp.text)["process_results"]
+        mapsets = {
+            mname: Mapset(mname, location_name, actinia, auth)
+            for mname in mapset_names
+        }
+        return mapsets
+
+    @classmethod
+    def create_mapset_request(cls, mapset_name, location_name, actinia, auth):
```

Comment:
Again, I would have done a private method rather than a classmethod. What exactly is the reason that you used a classmethod here?

Reply:
I commented on this in the above comment, but let me know if you need any more clarification.